File: scratch/test_parsing.js
import fetch from 'node-fetch'; // Wait, let's use a native fetch if node 18+, or a simpler HTTP request in Node

// Let's write a simpler script using native https to fetch and parse the CSV
import https from 'https';

const url = "https://docs.google.com/spreadsheets/d/e/2PACX-1vTj1EanSaF9L7EOt7uzJkaYM-4nSiJSe4cRG7Zp4oVai10WUghucUCUEhsHPcqnCFEAWyKLDD1AIyzB/pub?output=csv&gid=365647797";

function parseCSV(text) {
  const lines = [];
  let row = [];
  let inQuotes = false;
  let currentValue = '';
  
  for (let i = 0; i < text.length; i++) {
    const char = text[i];
    const nextChar = text[i + 1];
    
    if (char === '"') {
      if (inQuotes && nextChar === '"') {
        currentValue += '"';
        i++;
      } else {
        inQuotes = !inQuotes;
      }
    } else if (char === ',' && !inQuotes) {
      row.push(currentValue.trim());
      currentValue = '';
    } else if ((char === '\n' || char === '\r') && !inQuotes) {
      if (char === '\r' && nextChar === '\n') i++;
      row.push(currentValue.trim());
      lines.push(row);
      row = [];
      currentValue = '';
    } else {
      currentValue += char;
    }
  }
  if (currentValue || row.length > 0) {
    row.push(currentValue.trim());
    lines.push(row);
  }
  return lines;
}

function isIntercompanyTransfer(description) {
  if (!description) return false;
  const desc = description.toLowerCase();
  
  const isTransfer = desc.includes('traspaso') || desc.includes('transf') || desc.includes('transfe') || desc.includes('tef') || desc.includes('trasp');
  if (isTransfer) {
    const hasGmd = desc.includes('gmd') || desc.includes('grupo marketing digital') || desc.includes('grupo marketing');
    const hasGrafhika = desc.includes('grafhika') || desc.includes('copy center') || desc.includes('copycenter');
    if (hasGmd || hasGrafhika) return true;
  }
  
  const isRescate = desc.includes('rescate') && (desc.includes('fondo') || desc.includes('ffmm') || desc.includes('mutuo'));
  if (isRescate) return true;
  
  return false;
}

https.get(url, (res) => {
  let data = '';
  res.on('data', (chunk) => { data += chunk; });
  res.on('end', () => {
    const rows = parseCSV(data);
    console.log("Total rows:", rows.length);
    
    for (let i = 1; i < rows.length; i++) {
      const row = rows[i];
      if (row.length < 2) continue;
      const desc = row[1];
      if (desc.toLowerCase().includes('rescate')) {
        const isInter = isIntercompanyTransfer(desc);
        console.log(`Row ${i}: Desc='${desc}' | isIntercompanyTransfer=${isInter} | RawRow=${JSON.stringify(row)}`);
      }
    }
  });
}).on('error', (err) => {
  console.log("Error:", err.message);
});

File: scratch/test_parsing.py
import urllib.request
import csv
import io

url = "https://docs.google.com/spreadsheets/d/e/2PACX-1vTj1EanSaF9L7EOt7uzJkaYM-4nSiJSe4cRG7Zp4oVai10WUghucUCUEhsHPcqnCFEAWyKLDD1AIyzB/pub?output=csv&gid=365647797"

def is_intercompany_transfer(description):
    if not description:
        return False
    desc = description.lower()
    
    # 1. Detección de traspasos entre empresas hermanas (GMD y Grafhika)
    is_transfer = 'traspaso' in desc or 'transf' in desc or 'transfe' in desc or 'tef' in desc or 'trasp' in desc
    if is_transfer:
        has_gmd = 'gmd' in desc or 'grupo marketing digital' in desc or 'grupo marketing' in desc
        has_grafhika = 'grafhika' in desc or 'copy center' in desc or 'copycenter' in desc
        if has_gmd or has_grafhika:
            return True
            
    # 2. Detección de Rescates de Fondos Mutuos (movimientos internos de activos entre FFMM y Caja)
    is_rescate = 'rescate' in desc and ('fondo' in desc or 'ffmm' in desc or 'mutuo' in desc)
    if is_rescate:
        return True
        
    return False

try:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    with urllib.request.urlopen(req) as response:
        html = response.read().decode('utf-8')
    
    f = io.StringIO(html)
    reader = csv.reader(f)
    rows = list(reader)
    
    for idx, r in enumerate(rows[1:]):
        if len(r) < 2 or not r[1]:
            continue
        desc = r[1]
        if 'rescate' in desc.lower():
            res = is_intercompany_transfer(desc)
            print(f"Row {idx+1}: Desc='{desc}' | is_intercompany_transfer={res} | Monto={r[2]} | Tipo={r[3]}")
except Exception as e:
    print("Error:", e)
